### ISSUE TITLE
feature: Multiple indices in EntityStore

### ISSUE THREAD

**Issue by j2L4e** [datorama/akita#43]
Example: you might want to look up items by a tag that's attached to it. Say we have an EntityStore with `items` where each item can have one or several tags:

```
const items = [
  {id:1, tags: [1, 2]},
  {id:2, tags: [1]}
  {id:3, tags: [2, 3]}
];
```

The resulting State looks like this: 

```
{
  entities: {
    1: {id:1, tags: [1, 2]},
    2: {id:2, tags: [1]}
    3: {id:3, tags: [2, 3]}
  },
  ids: [1,2,3]
}
```

Filtering or finding items by tag is cumbersome and slow. It'd be nice to get the possibility of indexing items by one or more keys different from `id` to get a state similar to this one:

```
{
  entities: {
    1: {id:1, tags: [1, 2]},
    2: {id:2, tags: [1]}
    3: {id:3, tags: [2, 3]}
  },
  entitiesByTag: {
    1: [1, 2], // just the ids for brevity, actual entities should be here
    2: [1, 3], 
    3: [3]
  },
  ids: [1,2,3]
}
```

Implementing this reactively isn't feasible as you'd have to either diff or reindex the entire thing on every state change.

**Comment by NetanelBasal** [comment]
Actually you can do it easily with the `filterBy` option:

```ts
selectAll({ filterBy: entity => entity.tags.includes(tagId) })
```

As for the performance, I don't think there should be a significant problem. Browsers are highly optimized when it comes to `for` loops. Did you experience a performance issue?

**Comment by j2L4e** [comment]
Thanks. You are right, this might be a case of premature optimization. It does have its place though, when you handle larger amounts of normalized data, especially on less powerful clients.

**Comment by NetanelBasal** [comment]
Agree, but I think it's hard to develop something that will satisfy all cases.

**Comment by oleersoy** [comment]
There's [Relational Pouch](https://github.com/pouchdb-community/relational-pouch) so if the store is persisted to pouchdb it might be a good solution performance wise.  Hoping to get time to play with it soon.
